s3tests_boto3/functional/test_iam.py

```diff
@@ -862,3 +866,219 @@ def test_verify_allow_iam_actions():
     response = iam_client_alt.delete_user_policy(PolicyName='AllowAccessPolicy',
                                           UserName=get_alt_user_id())
     assert response['ResponseMetadata']['HTTPStatusCode'] == 200
+
+
+def nuke_user(client, name):
+    # delete access keys, user policies, etc
+    client.delete_user(UserName=name)
+
+def nuke_users(client, **kwargs):
+    p = client.get_paginator('list_users')
+    for response in p.paginate(**kwargs):
+        for user in response['Users']:
+            try:
+                nuke_user(client, user['UserName'])
+            except:
+                pass
+
+# fixture for iam account root user
+@pytest.fixture
+def iam_root(configfile):
+    client = get_iam_root_client()
```

Comment:
@cbodley : just curious - how do we differentiate between a root user and other IAM users in rgw?

Reply:
in https://github.com/ceph/ceph/pull/54333/commits/b90e93c9543721c4b3e6d870ff4754df2cf59fbf i extended enum RGWIdentityType with a TYPE_ROOT that's used when a user is created with the --account-root option
```c++
if (s->identity->get_identity_type() == TYPE_ROOT) { // account root user
```